server/TracyView.cpp

```diff
@@ -12269,9 +12269,29 @@ struct SrcLocZonesSlim
     int16_t srcloc;
     size_t numZones;
     int64_t total;
-    int64_t selfTotal;
 };
 
+void View::AccumulationModeComboBox()
+{
+    ImGui::SameLine();
+    ImGui::Spacing();
+    ImGui::SameLine();
+    ImGui::TextUnformatted( "Timing" );
+    ImGui::SameLine();
+    const char* accumulationModeTable = m_statMode == 0 ? "Self only\0With children\0Non-reentrant\0" : "Self only\0With children\0";
+    ImGui::SetNextItemWidth( ImGui::CalcTextSize( "Non-reentrant" ).x + ImGui::GetTextLineHeight() * 2 );
+    if ( m_statMode != 0 && m_statAccumulationMode == AccumulationMode::NonReentrantChildren )
+    {
+        m_statAccumulationMode = AccumulationMode::SelfOnly;
+    }
+    int accumulationMode = static_cast<int>( m_statAccumulationMode );
+    ImGui::Combo( "##accumulationMode", &accumulationMode, accumulationModeTable );
+    m_statAccumulationMode = static_cast<AccumulationMode>( accumulationMode );
+    ImGui::SameLine();
+    ImGui::Spacing();
+    ImGui::SameLine();
```

Comment:
I think it is invalid to force `SameLine` and `Spacing` here (and in the function prologue). It should be responsibility of the code calling this function to properly glue the item to its surroundings. Right now it seems that a `Separator` call was catched by `SameLine`:

![obraz](https://user-images.githubusercontent.com/600573/123521910-e4bfc900-d6b9-11eb-827a-b1d72c8a8d60.png)

Reply:
> I think it is invalid to force `SameLine` and `Spacing` here (and in the function prologue). It should be responsibility of the code calling this function to properly glue the item to its surroundings. Right now it seems that a `Separator` call was catched by `SameLine`:

Good eye.  How does the latest look?